conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanGenerator.java

```diff
@@ -420,13 +482,21 @@ private static ImmutableList<EnrichedField> createFields(TypeMapper typeMapper,
                 .collect(ImmutableList.toImmutableList());
     }
 
-    private static MethodSpec createConstructor(Collection<EnrichedField> fields, Collection<FieldSpec> poetFields) {
+    private static void addConstructor(TypeSpec.Builder builder, Collection<EnrichedField> fields) {
+        // there's no reason to keep `createConstructor` around, we can do everything in this method. I'm just making a
+        // call to `createConstructor` here for clarity.
+        builder.addMethod(createConstructor(fields));
+    }
+
+    private static MethodSpec createConstructor(Collection<EnrichedField> fields) {
         MethodSpec.Builder builder = MethodSpec.constructorBuilder().addModifiers(Modifier.PRIVATE);
 
-        Collection<FieldSpec> nonPrimitivePoetFields =
-                Collections2.filter(poetFields, f -> !Primitives.isPrimitive(f.type));
-        if (!nonPrimitivePoetFields.isEmpty()) {
-            builder.addStatement("$L", Expressions.localMethodCall("validateFields", nonPrimitivePoetFields));
+        Collection<EnrichedField> nonPrimitiveEnrichedFields = Collections2.filter(fields, f -> !f.isPrimitive());
+        if (!nonPrimitiveEnrichedFields.isEmpty()) {
+            builder.addStatement(
+                    "$L",
+                    Expressions.localMethodCall(
+                            "validateFields", EnrichedField.toPoetSpecs(nonPrimitiveEnrichedFields)));
```

Comment:
Not the point of this PR. Can ignore.
---

I removed `poetFields` as an argument, which requires me to call `EnrichedField.toPoetSpecs` and do an iteration over the `nonPrimitiveEnrichedFields` array. It's probably not a big performance hit(?), but I'm still leaning towards not introducing this change.